### PR TITLE
Refactor feat #75

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
close: #75

### CODE REVIEW

File: src/PlotReducer.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { ReducerActions, State } from './types';
```
Comment (maasencioh) [inline src/PlotReducer.tsx:1]: by convention this file should be called PlotReducer.ts, tsx is for jsx with typescript

File: src/types.ts [hunk 2]
```diff
@@ -135,9 +135,11 @@ export interface PlotContextType {
   axisContext: Record<string, AxisContextType>;
 }
 
+export type AxisType = Record<string, Omit<AxisParentProps, 'id'>>;
```
Comment (maasencioh) [inline src/types.ts:138]: is this type used somewhere else? if not you could skip the exportation

File: stories/spectra/tga.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      width={1400}
+      height={540}
+      margin={{ bottom: 45, left: 40, top: 40, right: 40 }}
+      viewportStyle={{ stroke: 'black' }}
+    >
+      <LineSeries
+        data={data1}
+        xAxis="x"
+        yAxis="y"
+        lineStyle={{ stroke: 'red' }}
```
Comment (maasencioh) [inline stories/spectra/tga.stories.tsx:24]: you could remove both strokes, by default is giving these colors https://github.com/d3/d3-scale-chromatic#schemeSet1

**Review by targos — APPROVED** [review]

**Review by maasencioh — COMMENTED** [review]
At the end of the blue curve you have a curve, is this supposed to be correct?
Screenshot 2021-01-20 100651

File: src/PlotReducer.tsx (hunk 1 above)
Comment (targos) [reply src/PlotReducer.tsx:1]: > At the end of the blue curve you have a curve, is this supposed to be correct?
> Screenshot 2021-01-20 100651

Yes, it is how the data is.

**Comment by targos** [comment]
> At the end of the blue curve you have a curve, is this supposed to be correct?
> Screenshot 2021-01-20 100651

Yes, it is how the data is.

**Pull request merged**